app/models/hud_reports/report_client_base.rb

```diff
@@ -4,11 +4,62 @@
 # License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
 ###
 
+# frozen_string_literal: true
+
 # Base class for a report type's clients.
 module HudReports
   class ReportClientBase < GrdaWarehouseBase
     self.abstract_class = true
 
     has_many :report_clients, as: :universe_membership, dependent: :destroy
+
+    def display_value(col, pii_policy:, include_content_tag: true)
+      # We are expecting some columns to come through with a dot notation.
+      # This will separate the object and method in order to call the appropriate column/method
+      # while similarly allowing the method to be called if the column is not passed using dot notation.
+      cell = col.to_s.split('.').inject(self) do |obj, method|
+        if obj.class.column_names.include?(method)
+          obj[method]
+        else
+          obj.send(method)
+        end
+      end
+
+      return ActionController::Base.helpers.content_tag(:pre, JSON.pretty_generate(cell)) if include_content_tag && (cell.is_a?(Array) || cell.is_a?(Hash))
+      return ModelApplicationHelper.new.yes_no(cell, include_content_tag: include_content_tag) if cell.in?([true, false])
+
+      case col.to_s
+      when /project_type$/
+        HudUtility2024.project_type_brief(cell)
+      when /prior_living_situation$/
+        HudUtility2024.living_situation(cell)
+      when /.*destination$/
+        HudUtility2024.destination(cell)
+      when /_days_/
+        number_with_delimiter(cell)
+      when /.*length_of_stay$/
+        HudUtility2024.residence_prior_length_of_stay(cell)
+      when /^ssn$/
+        ModelApplicationHelper.new.masked_ssn(cell, include_content_tag: include_content_tag)
```

Comment:
I think we're trying to retire masked ssn in favor of the pii provider pattern.
* Are we doing this because we only ever want masked ssns in the report so we don't want to obey the policy?
* It looks like `include_content_tag` would just add an enclosing span. Is that span necessary?

```suggestion
         GrdaWarehouse::PiiProvider.viewable_ssn(cell, policy: pii_policy)
```